chainstate/storage/src/internal/mod.rs

```diff
@@ -42,12 +42,35 @@ use crate::{
 mod store_tx;
 pub use store_tx::{StoreTxRo, StoreTxRw};
 
+mod version;
+pub use version::ChainstateStorageVersion;
+
 /// Store for blockchain data, parametrized over the backend B
 pub struct Store<B: storage::Backend>(storage::Storage<B, Schema>);
 
 impl<B: storage::Backend> Store<B> {
     /// Create a new chainstate storage
-    pub fn new(backend: B) -> crate::Result<Self> {
+    pub fn new(backend: B, chain_config: &ChainConfig) -> crate::Result<Self> {
+        let mut storage = Self::from_backend(backend)?;
+
+        // Set defaults if missing
+
+        if storage.get_storage_version()?.is_none() {
+            storage.set_storage_version(ChainstateStorageVersion::CURRENT as u32)?;
+        }
+
+        if storage.get_magic_bytes()?.is_none() {
+            storage.set_magic_bytes(chain_config.magic_bytes())?;
+        }
+
+        if storage.get_chain_type()?.is_none() {
+            storage.set_chain_type(chain_config.chain_type().name())?;
+        }
```

Comment:
Should we require an all-or-nothing storage state here? I.e. enforce storage version, magic bytes, chain type to be all `None` or all `Some(_)` and report an error is some of them are initialised but others are not.

Also the three updates could be done in a transaction to get atomic initialisation.

Reply:
Sounds like a good idea, but I wouldn't harp on it a lot, because if any of them isn't set, the whole thing will fail.

Reply:
used transaction